### ISSUE TITLE
Question: Is the `9p` executable required for navigating the tour?

### ISSUE THREAD

**Issue by legendre6891** [sminez/ad#51]
Hi - this is an interesting project and after watching the video in your blog, I'm curious to learn more. 

When I'm navigating the tour, I see this error message

```.../lib/ad.sh: line[xx] 9p command not found```

Evidently, this means that I should have a binary `9p` in my shell's PATH.  Is this binary supplied with the repo, or is this meant as a 3rd party runtime dependency that the user should install themselves?  Thanks.

**Comment by srenatus** [comment]
I'm not the owner, but I think I know the answer: Yes it is required. I got mine from [plan9port](https://github.com/9fans/plan9port/).

**Comment by sminez** [comment]
ah sorry @legendre6891, I really should call that out on the first slide of the tour. The `fsys` slide mentions it but nothing talks about how to install it I don't think. I've been wondering about allowing `ad` itself to be used as a 9p client via a flag so you don't need to have `9p` as a separate binary.

**Comment by izuzak** [comment]
@sminez 👋 Is there anything else required other than installing the 9p tools? 

I installed https://github.com/9fans/plan9port, and the `9p` tool is available and in PATH, and when I try to use "@" on "Slide" -- I get this error:

![Image](https://github.com/user-attachments/assets/8385c61b-045a-4808-be9e-f80bb9467e8c)

**Comment by sminez** [comment]
@izuzak ah, can you try exporting `NAMESPACE="/tmp/ns.$USER.$DISPLAY"` and trying again? I thought that was used by default by the `9p` tool but maybe not

**Comment by izuzak** [comment]
> ah, can you try exporting `NAMESPACE="/tmp/ns.$USER.$DISPLAY"` and trying again? I thought that was used by default by the `9p` tool but maybe not

@sminez Sure sure! I now get something else:

![Image](https://github.com/user-attachments/assets/55aed41d-f726-480f-89e7-de7957f6a183)

I'm on MacOS Sequoia 15.2, in case that's relevant. Apologies if I'm missing something obvious, and totally feel free to just tell me it's all in the docs for https://github.com/9fans/plan9port and I'll go read that 😸

**Comment by sminez** [comment]
No, it's me making assumptions about running under X11 I'm afraid. I've just double checked the ninep crate and there I'm not using `$DISPLAY`, I'm just hard coding `:0` it seems. So for now you'll need to export `NAMESPACE=/tmp/ns.$USER.:0` and that _should_ do it 🤞

**Comment by izuzak** [comment]
> No, it's me making assumptions about running under X11 I'm afraid. I've just double checked the ninep crate and there I'm not using `$DISPLAY`, I'm just hard coding `:0` it seems. So for now you'll need to export `NAMESPACE=/tmp/ns.$USER.:0` and that _should_ do it 🤞

Yay, that fixes things indeed. 👍 Many thanks!
